### PR TITLE
261 booking request tracker

### LINKED ISSUE #261: Booking Request Tracking
### User story
As a system approver, I can view a history of the changes made to a booking as well as the current status of the booking in the approval process. 

### Narrative
Any user can see, in log form, the list of changes made to the booking. History should be viewable to all users who have a pending booking request. Users can also see a progress bar that displays the booking's current state in the approval process.

### Acceptance criteria

| # | Given | When | Then |
|---|-------|------|------|
|1|I am a user|I wish to see the changes made to a booking in the approval process|I can see a log of changes made to the booking request.|
|2|I am a user|I wish to see my booking's status in the approval process.|I can view a status bar that reflects the booking's current state.|

### Mockup

Booking Tracking

### PR DESCRIPTION
**Pull request by KeeganWoods1**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**
Please provide enough information so that others can review your pull request:

Allows users to see a log of any changes that occurred to a given booking.

Explain the **details** for making this change. What existing problem does the pull request solve?

It is required that system users be able to see the booking status as well as a changelog.

### Test plan (required)
Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

### Closing issues
List all issues this PR resolves.
- `closes #261 ` 

### Pre-Merge checklist 
Complete these items before requesting reviews
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions 

### Sub-tasks Checklist
- [x] UI Implementation
- [x] UI Test
- [x] Backend Implementation
- [x] Backend Tests
- [ ] Integration Tests
- [ ] System Tests
- [ ] Documentation

### CODE REVIEW

**Review by EvanDime — COMMENTED** [review]
Everything seems to be fine. I will approve once this branch is up to date with main. (ping me when you do)

**Review by EvanDime — APPROVED** [review]

**Comment by EvanDime** [comment]
@KeeganWoods1 
Sorry this branch needs to be updated and revalidated. Maybe tag everyone who needs to review or multiple people since this is teh 2nd time it is falling behind

File: app/Events/BookingRequestUpdated.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    /**
+     * Get the channels the event should broadcast on.
+     *
+     * @return \Illuminate\Broadcasting\Channel|array
+     */
+    public function broadcastOn()
+    {
+        return new PrivateChannel('channel-name');
```
Comment (danseremet) [inline app/Events/BookingRequestUpdated.php:41]: Does this have the correct name?

**Review by danseremet — APPROVED** [review]
Looks like it works :+1: 
I'm approving, but check my small comment below though.
image

**Pull request merged**